### ISSUE TITLE
Make kubectl-validate compatible with WASM

### ISSUE THREAD

**Issue by eddycharly** [kubernetes-sigs/kubectl-validate#79]
Currently it's not possible to use kubectl-validate and build a wasm module because of dependencies that bring a reference to etcd (that has non supported syscalls).

```
GOOS=js GOARCH=wasm go build -o ./main.wasm ./main.go

package command-line-arguments
        imports sigs.k8s.io/kubectl-validate/pkg/cmd
        imports k8s.io/apiextensions-apiserver/pkg/apiserver
        imports k8s.io/apiextensions-apiserver/pkg/registry/customresource
        imports k8s.io/apiserver/pkg/registry/generic
        imports k8s.io/apiserver/pkg/storage/storagebackend/factory
        imports go.etcd.io/etcd/client/pkg/v3/transport
        imports golang.org/x/sys/unix: build constraints exclude all Go files in /Users/charled/Documents/dev/eddycharly/kubectl-validate/vendor/golang.org/x/sys/unix
```

**Comment by alexzielenski** [comment]
Yeah this is mainly due to our reliance on `apiextensions-apiserver` to validate CRDs. THey ahve recursive schemas which the k8s validators choke on.

If we remove the ability to validate CRDs by schema and just skip/error on CRDs then this requirement can be removed.

**Comment by eddycharly** [comment]
@alexzielenski you suggest to remove the CRD support ?

**Comment by eddycharly** [comment]
IIRC code like this was part of the problem `return rest.BeforeCreate(strat, request.WithNamespace(context.TODO(), ""), obj)`

**Comment by alexzielenski** [comment]
> @alexzielenski do you suggest removing CRD support ?

I am waffling about it. It is the only resource we cheat for validation on and it would be a lot of work to support its recursive schema. I wonder if users would miss it


> ike this was part of the problem `return rest.BeforeCreate(strat, request.WithNamespace(context.TODO(), ""), obj)`

Ah, if that is the case that can also be quite easily removed & replicated within our own codebase

**Comment by alexzielenski** [comment]
Took another look at this. We rely on a lot more code than just ValidateCustomResourceDefinition inside apiextensions-apiserver. Unfortunately itd be quite a large refactor to move them out of that package upstream, and I would rather keep the dependency to ensure the code says the same.
